src/libs/pois.js

```diff
@@ -12,16 +12,32 @@ export function toUrl(poi) {
   return poi.name ? `${poi.id}@${slug(poi.name)}` : poi.id;
 }
 
-export async function getInputValue(poi) {
-  if (poi.type === 'latlon' || !poi.name) {
-    return await fetchAddressLabel(poi);
+function formatAddress(name, city, country) {
+  return [name, city, country]
+    .filter(i => i) // Filter out any undefined value
+    .join(', ');
+}
+
+export async function getNameAddress(poi) {
+  if (poi.type === 'latlon') {
+    // latlon poi do not have human readable name,
+    // so return street address instead
+    return getStreetAddress(poi);
   }
-  return poi.getInputValue();
+
+  poi = await fetchAddress(poi);
+  return formatAddress(poi.name, poi.address?.city, poi.address?.country);
+}
+
+export async function getStreetAddress(poi) {
+  poi = await fetchAddress(poi);
+  return formatAddress(poi.address?.name, poi.address?.city, poi.address?.country);
 }
 
-async function fetchAddressLabel(poi) {
-  const address = await IdunnPoi.poiApiLoad(poi);
-  return address.alternativeName || address.name;
+async function fetchAddress(poi) {
```

Comment:
It's a bit strange to return a poi in a function named `fetchAddress`. What do you think about returning the `address` only here?
And maybe if an API call is needed, store the address in the `poi` argument in the function, for caching… But maybe it's a bit too side-effect, I don't know.

Reply:
You right, I update and now only return the address.
Do you think we can safely assign the address to the poi for caching purpose ?  I'm not sure either